app/controllers/groupings_controller.rb

```diff
@@ -1,26 +1,31 @@
-class Admin::LayoutsController < ApplicationController
-  before_action :load_grouping, only: %i(edit update)
+class GroupingsController < ApplicationController
+  before_action :load_grouping, only: %i(show edit update)
+
+  def show
+    @parent = @grouping.parent_grouping
+    @grouping_name = @grouping.name
+    @children = @grouping.children
+  end
 
   def edit
     @grouping_name = @grouping.name
-    @desks = @grouping.desks
-    @child_groupings = @grouping.child_groupings
+    @children = @grouping.children
   end
 
   def update
     lay_out = JSON.parse(params.require(:layout))
     @grouping.children.each do |area|
-      x, y = lay_out[area.class.name.downcase][area.id.to_s].values_at('x', 'y')
-      if area.coordinate.nil?
-        area.coordinate = Coordinate.create(x: x, y: y)
-        render_exception(area) unless area.save
-      else
+      x, y = lay_out[area.type][area.id.to_s].values_at('x', 'y')
```

Comment:
See here the use of `grouping#type` and `desk#type` to index the layout from the client.